spec.bs

```diff
@@ -7284,6 +7343,31 @@ for a given [=k-anonymity key=]. These records are stored in the [=user agent=].
   :: The [=moment=] when the k-anonymity status in this record was last fetched.
 </dl>
 
+<h3 id=auction-data-config-header>auction data config</h3>
+
+An <dfn>auction data config</dfn> is a [=struct=] with the following [=struct/items=]:
+
+<dl dfn-for="auction data config">
+  :  <dfn>seller</dfn>
+  :: An [=origin=].
+    The origin of the seller running the ad auction. The [=origin/scheme=] must be "`https`".
+  :  <dfn>coordinator</dfn>
+  :: The origin of the coordinator hosting public encryption keys for the server
+    running the ad auction. The [=origin/scheme=] must be "`https`".
+  :  <dfn>request size</dfn>
+  :: {{unsigned long}} or null. An optional field, containing the desired size
+    for the returned encrypted request blob.
+  :  <dfn>per buyer config</dfn>
+  :: A [=map=] whose [=map/keys=] are [=origins=] and [=map/values=] are [=auction data buyer config=].
+</dl>
+
+An <dfn>auction data buyer config</dfn> is a [=struct=] with the following [=struct/items=]:
+<dl dfn-for="auction data buyer config">
+  :  <dfn>size</dfn>
+  :: {{unsigned long}} or null. An optional field, initially null, containing the

```

Comment:
as discussed offline, we may want to add something like "when [=auction data config/request size=] is null, it's required and cannot be null".

Reply:
That doesn't work though, because the [=auction data config/request size=] is never null when [=auction data config/per buyer config=] [=map/is not empty=]. The real constraint is on {{AdAuctionDataConfig}} and {{AdAuctionDataBuyerConfig}}.

Is there somewhere to document the constraints for IDL arguments?

Reply:
You can use "domintro" like this: https://github.com/WICG/turtledove/blob/main/spec.bs#L5960. It renders as https://wicg.github.io/turtledove/#dom-reportingbrowsersignals-topwindowhostname

Reply:
"domintro" is exactly what I was looking for. Thanks for pointing that out.

I added "domintro" describing the arguments to `navigator.getInterestGroupAdAuctionData()`.